consensus/scp/src/msg.rs

```diff
@@ -30,6 +30,22 @@ pub struct NominatePayload<V: Value> {
     pub Y: BTreeSet<V>,
 }
 
+impl<V: Value> NominatePayload<V> {
+    /// Create a NominatePayload
+    ///
+    /// # Arguments
+    /// * `x` - Values voted nominated.
+    /// * `y` - Values accepted nominated.
+    ///
+    /// `x` and `y` must be disjoint sets.
+    pub fn new(x: &HashSet<V>, y: &HashSet<V>) -> Self {
```

Comment:
Syntactic sugar.

Under the hood, NominatePayload still uses BTreeSets for X and Y so that Nominate messages have a unique encoding. However, the actual ordering of X and Y is meaningless.